docs/understanding-microsoft.spark.md

```diff
@@ -0,0 +1,33 @@
+
+# Components Overview
+
+The following component diagram depicts a high-level overview of the vital components participating in a .NET for Apache Spark application lifecycle:
+
+![Icon](img/Spark-dotnet-integration-component-diagram.png)
+
+This diagram illustrates the interaction between the Apache Spark components and the .NET components. The `Microsoft.Spark` NuGet package contains a number of wrappers over the Scala JVM objects (with reference to internal objects mapped in the bridge), and allows calling various methods on those objects. Unless `Collect()` is called on a dataframe, no data is actually loaded into the .NET app; the solution acts as a proxy for the Scala Spark API. Key components include:
```

Comment:
Why does `Collect()` function retrieval data to .Net? How about UDF functions here?

Reply:
Thanks for highlighting the unclear part! 
 What I meant is that this is the only way to load data on executed on a driver `User .NET App`. 

![image](https://github.com/user-attachments/assets/d1286c48-fe68-4321-98e1-c68b3dd73ec8)

For the .NET worker, we can send information in two ways: Broadcast and UDFs. I’ll clarify the wording here.